Kickstarter-iOS/Views/Cells/SettingsTextInputCell.swift

```diff
@@ -0,0 +1,119 @@
+import Prelude
+import UIKit
+
+private struct Margin {
+  static let leftRight: CGFloat = 8
+  static let topBottom: CGFloat = 15
+}
+
+class SettingsTextInputCell: UITableViewCell {
+  // MARK: - Accessors
+
+  private lazy var stackView: UIStackView = {
+    let layoutMargins = UIEdgeInsets.init(topBottom: Margin.topBottom, leftRight: Margin.leftRight)
+    return UIStackView()
+      |> \.isLayoutMarginsRelativeArrangement .~ true
+      |> \.layoutMargins .~ layoutMargins
+      |> \.spacing .~ 8
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  private lazy var label: UILabel = {
+    return UILabel()
+      |> \.adjustsFontForContentSizeCategory .~ true
+      |> \.font .~ UIFont.preferredFont(forTextStyle: .body)
+      |> \.numberOfLines .~ 0
+  }()
+
+  private lazy var textField: UITextField = {
+    return UITextField()
+      |> \.adjustsFontForContentSizeCategory .~ true
+      |> \.font .~ UIFont.preferredFont(forTextStyle: .body)
+      |> \.textAlignment .~ .right
+      |> \.isSecureTextEntry .~ true
+  }()
+
+  public func configure(with title: String, placeholder: String) {
+    _ = self.label
+      |> \.text .~ title
+
+    _ = self.textField
+      |> \.placeholder .~ placeholder
+  }
+
+  // MARK: - Lifecycle
+
+  override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+    super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+    self.stackView.addArrangedSubview(self.label)
+    self.stackView.addArrangedSubview(self.textField)
+
+    self.contentView.addSubview(self.stackView)
+    self.stackView.constrainEdges(to: self.contentView)
+    self.stackView.heightAnchor.constraint(greaterThanOrEqualToConstant: 44).isActive = true
+
+    self.label.setContentHuggingPriority(.required, for: .horizontal)
+    self.label.setContentCompressionResistancePriority(.required, for: .horizontal)
+
+    self.textField.setContentHuggingPriority(.defaultLow, for: .horizontal)
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: - Layout
+
+  override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+    super.traitCollectionDidChange(previousTraitCollection)
+
+    let traitCollection = self.traitCollection
+
+    if previousTraitCollection?.preferredContentSizeCategory != traitCollection.preferredContentSizeCategory {
+      self.configure(with: traitCollection)
+    }
+  }
+
+  private func configure(with traitCollection: UITraitCollection) {
```

Comment:
This is the key to the Dynamic Type resizing functionality.

When the category becomes too big (accessible) we switch from horizontal alignment to vertical alignment and vice versa. This allows placing `label` and `text field` above each other or side by side (depending on the size category)